src/components/BottomMenuModal/SendModal/SendModalTokensTabView.tsx

```diff
@@ -103,6 +127,218 @@ const SendModalTokensTabView = ({ payload }: { payload?: SendModalData }) => {
+            .filter(
+              (value): value is TokenAssetSelectOption => value !== undefined
+            );
+          if (feeOptions && feeOptions.length > 0 && feeOptions[0]) {
+            setFeeAssetOptions(feeOptions);
+            // get Skandha gas price
+            getGasPrice(selectedAsset.chainId).then((res) => {
+              setGasPrice(res);
+            });
+            setSelectedFeeAsset({
+              token: feeOptions[0].asset.contract,
+              decimals: feeOptions[0].asset.decimals,
+            });
+            setSelectedPaymasterAddress(feeOptions[0].id.split('-')[2]);
+            setPaymasterContext({
+              mode: 'commonerc20',
+              token: feeOptions[0].asset.contract,
+            });
+            setIsPaymaster(true);
+          } else {
+            setFeeType([]);
+          }
+        } else {
+          setPaymasterContext(null);
+          setIsPaymaster(false);
+          setFeeType([]);
+        }
+      }
+    );
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedAsset, walletPortfolio]);
+
+  const setApprovalData = async (gasCost: number) => {
+    if (selectedFeeAsset && gasPrice) {
+      const estimatedCost = Number(
+        utils.formatEther(BigNumber.from(gasCost).mul(gasPrice))
+      );
+      const costAsFiat = +estimatedCost * nativeAssetPrice;
+      const feeTokenPrice = await getPrices([selectedFeeAsset.token]);
+      let estimatedCostInToken;
+      if (feeTokenPrice) {
+        const feeTokenPriceInUSD = feeTokenPrice[0].usd;
+        estimatedCostInToken = (costAsFiat / feeTokenPriceInUSD).toFixed(
+          selectedFeeAsset.decimals
+        );
+        setFeeMin(estimatedCostInToken);
+        setApproveData(
+          encodeFunctionData({
+            abi: erc20Abi,
+            functionName: 'approve',
+            args: [
+              selectedPaymasterAddress as Address,
+              parseUnits(estimatedCostInToken, selectedFeeAsset.decimals),
+            ],
+          })
+        );
+      }
+    }
+  };
+
+  useEffect(() => {
+    if (!selectedAsset) return;
+    if (!gasPrice) return;
+    let gasCost = 0;
+    /*
+     * The gas cost is estimated based on the type of asset being sent.
+     * The gas cost is calculated based on the type of asset being sent
+     * and the chain ID of the asset. The gas cost is then used to calculate
+     * the estimated cost in fiat.
+     * And the token cost is generally estimated as $0.01 even for undeployed wallet
+     * though it can be even lower for deployed wallet to save rpc call for checking
+     * deployed wallet or not, we can use the same gas cost for both deployed and undeployed wallet
+     */
+    if (selectedAsset.chainId === 42161) {
+      if (selectedAsset.type === 'token') {
+        if (selectedAsset.asset.contract === ethers.constants.AddressZero) {
+          gasCost = Number(
+            process.env.REACT_APP_NATIVE_GAS_CONSUMPTION_ARBITRUM
```

Comment:
Do the gas consumtion values need to be an environment variable as opposed to a const definition?

Reply:
Yes since if its const you would need code change to change every time rather you can just change the env and restart if you want to change the values in the middle